plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialHalf.java

```diff
@@ -2,73 +2,66 @@
 
 import com.denizenscript.denizen.objects.LocationTag;
 import com.denizenscript.denizen.objects.MaterialTag;
-import com.denizenscript.denizencore.objects.core.ElementTag;
 import com.denizenscript.denizencore.objects.Mechanism;
-import com.denizenscript.denizencore.objects.ObjectTag;
-import com.denizenscript.denizencore.objects.properties.Property;
+import com.denizenscript.denizencore.objects.core.ElementTag;
 import com.denizenscript.denizencore.objects.properties.PropertyParser;
 import org.bukkit.block.BlockFace;
 import org.bukkit.block.data.Bisected;
 import org.bukkit.block.data.BlockData;
-import org.bukkit.block.data.Directional;
 import org.bukkit.block.data.type.Bed;
 import org.bukkit.block.data.type.Chest;
 import org.bukkit.util.Vector;
 
-public class MaterialHalf implements Property {
+public class MaterialHalf extends MaterialProperty<ElementTag> {
+
+    // <--[property]
+    // @object MaterialTag
+    // @name half
+    // @input ElementTag
+    // @description
+    // Controls the current half for a bisected material (like a door, double-plant, chest, or a bed).
+    // For "Bisected" blocks (doors/double plants/...), values are either "BOTTOM" or "TOP".
+    // For beds, values are either "HEAD" or "FOOT".
+    // For chests, values are either "LEFT" or "RIGHT".
+    // For the half name, see <@link tag MaterialTag.half>.

```

Comment:
Isn't that just linking to itself? should already automatically be in the mechanism's tag link